### ISSUE TITLE
[BUG] replicas removed and volume left in inconsistent state

### ISSUE THREAD

**Issue by himslm01** [longhorn/longhorn#4985]
### TLDR

A Harvester VM was running. At some point the root disk volume of the VM has transitioned into an inconsistent state in Longhorn causing the VM to fail. 

### Longer description
From [Slack conversation](https://rancher-users.slack.com/archives/CC2UQM49Y/p1669639403989499)

I have a 3 node Harvester v1.1.0 cluster which includes Longhorn v1.3.2

Hopefully unrelated, but two days prior to this incident two of the three nodes were re-build from scratch by draining one node, removing the node from the cluster, rebuilding the node, re-adding the node to the cluster, waiting for all of the Longhorn volumes to replicate across to the node, draining the second node, etc.

Two days later one of the VMs became unavailable. Investigating in the Lonhgorn UI I see that the root Volume had lost two replicas (not both on the re-built nodes).

![Lonhgorn Volume UI](https://user-images.githubusercontent.com/1175979/205265207-8938592a-5f67-44c9-8524-4036a663d613.png)

Following @derekbit 's instructions I provided the contents of the volume’s replica folder.

```console
/var/lib/harvester/defaultdisk/replicas/pvc-3b137621-8434-456b-8f08-410ade838d8b-0ade9458 # ls -la
total 487332
drwx------  2 root root        4096 Nov 28 05:12 .
drwxr-xr-x 30 root root        4096 Nov 28 05:30 ..
-rw-------  1 root root        4096 Nov 28 05:12 revision.counter
-rw-r--r--  1 root root 64424509440 Nov 28 05:12 volume-head-006.img
-rw-r--r--  1 root root         178 Nov 28 05:12 volume-head-006.img.meta
-rw-r--r--  1 root root 64424509440 Nov 28 05:12 volume-snap-0b1e6133-4c31-4b2d-a15d-3742e757fa76.img
-rw-r--r--  1 root root 64424509440 Nov 28 05:12 volume-snap-b5d617df-075c-4850-a495-7818238ae609.img
-rw-r--r--  1 root root         178 Nov 28 05:12 volume-snap-b5d617df-075c-4850-a495-7818238ae609.img.meta
-rw-r--r--  1 root root         195 Nov 28 05:12 volume.meta
```
And the contents of the three `.meta` files.

```console
harvester002:/var/lib/harvester/defaultdisk/replicas/pvc-3b137621-8434-456b-8f08-410ade838d8b-0ade9458 # cat volume-head-006.img.meta
{"Name":"volume-head-002.img","Parent":"volume-snap-b5d617df-075c-4850-a495-7818238ae609.img","Removed":false,"UserCreated":false,"Created":"2022-11-28T05:12:48Z","Labels":null}
```

```console
harvester002:/var/lib/harvester/defaultdisk/replicas/pvc-3b137621-8434-456b-8f08-410ade838d8b-0ade9458 # cat volume-snap-b5d617df-075c-4850-a495-7818238ae609.img.meta
{"Name":"volume-head-001.img","Parent":"volume-snap-0b1e6133-4c31-4b2d-a15d-3742e757fa76.img","Removed":false,"UserCreated":false,"Created":"2022-11-28T05:12:48Z","Labels":null}
```

```console
harvester002:/var/lib/harvester/defaultdisk/replicas/pvc-3b137621-8434-456b-8f08-410ade838d8b-0ade9458 # cat volume.meta
{"Size":64424509440,"Head":"volume-head-006.img","Dirty":false,"Rebuilding":true,"Error":"","Parent":"volume-snap-b5d617df-075c-4850-a495-7818238ae609.img","SectorSize":512,"BackingFilePath":""}
```

## To Reproduce

Unclear.

## Expected behavior

Don't lose replicas ;-)

## Log or Support bundle

Support Bundle: [longhorn-support-bundle_41f2cfe5-e432-4e6a-9f8c-bfab0577ae9b_2022-11-30T08-49-50Z.zip](https://github.com/longhorn/longhorn/files/10139884/longhorn-support-bundle_41f2cfe5-e432-4e6a-9f8c-bfab0577ae9b_2022-11-30T08-49-50Z.zip)

## Environment

 - Longhorn version: v1.3.2
 - Installation method: Harvester
 - Kubernetes distro and version: Harvester v1.1.0
   - Number of management node in the cluster: 3
   - Number of worker node in the cluster: 0
 - Node config
   - OS type and version: Harvester
   - CPU per node: 8
   - Memory per node: 64GB
   - Disk type(e.g. SSD/NVMe): NVMe (2TB)
   - Network bandwidth between the nodes: 2.5Gbps
 - Underlying Infrastructure: Baremetal
 - Number of Longhorn volumes in the cluster: 34
